v2/api/network/customizations/load_balancer_extension.go

```diff
@@ -120,33 +113,137 @@ func fuzzySetInboundNatRules(lb genruntime.ARMResourceSpec, inboundNatRules []ge
 
 	inboundNatRulesField := propertiesValue.FieldByName("InboundNatRules")
 	if (inboundNatRulesField == reflect.Value{}) {
-		return errors.Errorf("couldn't find InboundNatRules field on LoadBalancer")
+		return reflect.Value{}, errors.Errorf("couldn't find InboundNatRules field on LoadBalancer")
 	}
 
 	if inboundNatRulesField.Type().Kind() != reflect.Slice {
-		return errors.Errorf("inboundNatRules field on LoadBalancer was not of kind Slice")
+		return reflect.Value{}, errors.Errorf("InboundNatRules field on LoadBalancer was not of kind Slice")
+	}
+
+	return inboundNatRulesField, nil
+}
+
+func getNameField(natValue reflect.Value) (ret reflect.Value, err error) {
+	defer func() {
+		if x := recover(); x != nil {
+			err = errors.Errorf("caught panic: %s", x)
+		}
+	}()
+
+	nameField := natValue.FieldByName("Name")
+	if (nameField == reflect.Value{}) {
```

Comment:
There's a method for checking whether a Value is the zero value or not. Confusingly, it's not `IsZero()` - that's different.

```suggestion
	if nameField.IsValid() {
```

Reply:
Done and elsewhere in the shared `getChildCollectionField` function as well